000-vip-init.php

```diff
@@ -265,4 +265,28 @@
 	add_filter( 'wp_sitemaps_enabled', '__return_false' );
 }
 
+// We use HyperDB, so DB_* constants are not automatically defined
+if ( ! defined( 'DB_NAME' ) && ! defined( 'DB_HOST' ) && ! defined( 'DB_PASSWORD' ) && ! defined( 'DB_NAME' ) ) {
+	global $db_servers;
+	if ( ! is_array( $db_servers ) ) {
+		return;
+	}
+
+	$writable_servers = array_filter( $db_servers, function( $db_server ) {
+		return isset( $db_server[5] ) && $db_server[5] > 0;
```

Comment:
```suggestion
		return ! isset( $db_server[5] ) || $db_server[5] > 0;
```

Per the [documentation](https://github.com/Automattic/HyperDB/blob/trunk/db-config.php#L81-L82), this parameter is optional and is considered to be `1` if not defined.